day_3/story_game.py

```diff
@@ -1,9 +1,74 @@
-from scenario import SCENARIO
+from scenario import GREETING, SCENARIO
 
 
 class Node:
-    def __init__(self, position, left, right, text):
+    def __init__(self, position, left_pos, left, right_pos, right, text):
         self.position = position
+        self.left_event = left_pos
         self.left = left
+        self.right_event = right_pos
         self.right = right
         self.text = text
+
+
+class Story:
+    def __init__(self, story_dict: dict, greeting: str = ''):
+        self.story_dict = story_dict
+        self.greeting = greeting
+        self.story = {}
+
+    def make_story(self):
+        for position, details in self.story_dict.items():
+            node = Node(
+                position=position,
+                left_pos=details['left_pos'],
+                left=details['left'],
+                right_pos=details['right_pos'],
+                right=details['right'],
+                text=details['text'],
+            )
+            self.story[position] = node
+        event_codes = self.story.keys()
+        for event_code, event in self.story.items():
+            if event.left_event in event_codes:
+                event.left_event = self.story[event.left_event]
+            if event.right_event in event_codes:
+                event.right_event = self.story[event.right_event]
```

Comment:
It feels like we should be failing here, if there is no event in `event_codes`, since this actually means that the store has an invalid option somewhere.

Reply:
Sorry, but I do not understand what you mean. May I ask you to say it in other words?

Reply:
In current implementation, if there is some `left_event`, but it's not described in `event_codes` - we'll do nothing about it, and some node will have a faulty behavior, that's what caused `AttributeError: 'str' object has no attribute 'left_event'` that I've showed you. I think we should get an exception, if there is some value in `left_event` but this value is not in `event_codes`. I.g.:
```python
if event.left_node:
    event.left_event = self.story[event.left_event]  # will cause KeyError if there is no event with the required code
```